packages/fuel-indexer-lib/src/graphql/mod.rs

```diff
@@ -32,6 +39,362 @@ fn inject_native_entities_into_schema(schema: &str) -> String {
+                            internal_fields.push(connection_field);
+                        }
+                    }
+                }
+
+                obj.fields.append(&mut internal_fields);
+            }
+        }
+    }
+
+    ast.definitions.append(&mut pagination_types);
+
+    ast
+}
+
+fn create_edge_type_for_list_field(
+    list_field: &Positioned<FieldDefinition>,
+) -> TypeSystemDefinition {
+    let (base_type, name) = if let BaseType::List(t) = &list_field.node.ty.node.base {
+        if let BaseType::Named(n) = &t.base {
+            (t, n)
+        } else {
+            unreachable!("Edge type creation should not occur for non-list fields")
+        }
+    } else {
+        unreachable!("Edge type creation should not occur for non-list fields")
+    };
+
+    let edge_obj_type = ObjectType {
+        implements: vec![],
+        fields: vec![
+            Positioned::position_node(
+                list_field,
+                FieldDefinition {
+                    description: None,
+                    name: Positioned::position_node(list_field, Name::new("node")),
+                    arguments: vec![],
+                    ty: Positioned::position_node(
+                        list_field,
+                        Type {
+                            base: base_type.base.clone(),
+                            nullable: false,
+                        },
+                    ),
+                    directives: vec![],
+                },
+            ),
+            Positioned::position_node(
+                list_field,
+                FieldDefinition {
+                    description: None,
+                    name: Positioned::position_node(list_field, Name::new("cursor")),
+                    arguments: vec![],
+                    ty: Positioned::position_node(
+                        list_field,
+                        Type {
+                            base: BaseType::Named(Name::new("String")),
+                            nullable: false,
+                        },
+                    ),
+                    directives: vec![],
+                },
+            ),
+        ],
+    };
+
+    TypeSystemDefinition::Type(Positioned::position_node(
+        list_field,
+        TypeDefinition {
+            extend: false,
+            description: None,
+            name: Positioned::position_node(
+                list_field,
+                Name::new(format!("{}Edge", name)),
+            ),
+            directives: vec![Positioned::position_node(
+                list_field,
+                ConstDirective {
+                    name: Positioned::position_node(list_field, Name::new("internal")),
+                    arguments: vec![],
+                },
+            )],
+            kind: TypeKind::Object(edge_obj_type),
+        },
+    ))
+}
+
+fn create_connection_type_def_for_list_entity(name: &Name) -> TypeSystemDefinition {
+    let dummy_position = Pos {
+        line: 999,
```

Comment:
I see these are dummy values but maybe `usize::MAX` (or whatever value it accepts) to avoid magic numbers?

Reply:
Whoops, you're totally right; this was definitely a placeholder value.